singlestoredb/mysql/accel.c

```diff
@@ -1283,14 +1283,19 @@ static PyObject *read_row_from_packet(
                 case MYSQL_TYPE_LONG:
                 case MYSQL_TYPE_LONGLONG:
                 case MYSQL_TYPE_INT24:
+                {
+                    char *substr = calloc(out_l + 1, sizeof(char));
```

Comment:
I'm wondering if rather than copying the data every time if something like this might work. `data` contains the remaining data in the row, so if `data_l` is non-zero it means there are still characters at the end of the row buffer that can be swapped during parsing, then swapped back afterward. Whatever the fix is for this, would also need to be done in other places that use `strtoull`, `strtoll`, and `strtod`. The setting of `end` could go at the top of the loop for all cases. 
```
char end = out[out_l];
if (data_l) out[out_l] = '\0';
... parsing code ...
if (data_l) out[out_l] = end;
```

Reply:
Thank you for your suggestions. I refactored the code without copying the data and fixed other uses of `strto*` too.